write-rustdoc-hide-lines/src/formatter.rs

```diff
@@ -0,0 +1,84 @@
+use std::{
+    fs::{self, File},
+    io::{self, BufRead},
+    path::Path,
+};
+
+use crate::{
+    code_block_definition::CodeBlockDefinition, hidden_ranges::get_hidden_ranges,
+    utils::visit_dir_md_files,
+};
+
+pub fn run(dir: &Path) {
+    println!("Formatting folder: {:?}", dir);
+
+    let result = visit_dir_md_files(dir, &|entry| {
+        println!("{:?}", entry.path());
+
+        let file = File::open(entry.path())?;
+        let file_len = file.metadata().unwrap().len().try_into().unwrap();
+        let mut contents = String::with_capacity(file_len);
+        let mut is_inside_rust_code_block = false;
+        let mut rust_block: Vec<String> = vec![];
+
+        for line in io::BufReader::new(file).lines() {
```

Comment:
I think this loop could be extracted to a function that way this could have a doc test explaining what this whole app does with an actual example

Reply:
Thanks for the suggestion, I'll look into it.

Reply:
Resolved, see the other comment for more.